### PR TITLE
Mentor walkthrough refactor

### LINKED ISSUE #19: Inline code into walkthrough notes
Presenting the code to type would be easier with inline code, and separate explanations in the notes file.

### PR DESCRIPTION
**Pull request by macintoshhelper**
Fixes #19 by inlining code in mentor notes. Should make it easier for mentors to reproduce the walkthrough and explain everything better, while having the code ready.

Merge `mentor-walkthrough` first

### CODE REVIEW

File: walkthroughSteps/code-along-mentor-notes.md [hunk 1]
```diff
@@ -3,56 +3,225 @@
...
-  - we then `COMMIT;` our database - to confirm and execute the entire transaction.
+1. Create a folder inside the root project: `database`.
+    - In the folder `database`, we're going to setup the structure/schema of our database, and hardcoded data.
+
+2. Create a new file: `database/db_build.sql`.
+    1. Put a `BEGIN;` at the start of the file and a `COMMIT;` at the end of the file.
+        - Clarify that the init queries should be written between these lines (so BEGIN and COMMIT wrap around them).
+        - The code inside it is a transaction, so the multiple queries are run as one query/chunk. If an error occurs inside of here, it will rollback the previous commands, preventing messing up the database. Can think of it as SQL error handling
+
+    2. Write `DROP TABLE IF EXISTS superheroes cascade;`.
```
Comment (shiryz) [inline walkthroughSteps/code-along-mentor-notes.md]: cascade is a sql key word, better capitalized.

File: walkthroughSteps/code-along-mentor-notes.md [hunk 2]
```diff
@@ -3,56 +3,225 @@
...
+        ```sql
+        CREATE TABLE superheroes (
+          id SERIAL PRIMARY KEY,
+          name VARCHAR(100) NOT NULL,
+          superPower TEXT NOT NULL,
+          weight INTEGER DEFAULT 100
+        );
+        ```
+    
+        - All tables should have an integer `id` that is set as a `PRIMARY KEY` - this is used relate databased together (integer PRIMARY KEY helps with performance)
```
Comment (shiryz) [inline walkthroughSteps/code-along-mentor-notes.md:45]: Can you add a note that primary key is the combination of UNIQUE and NOT NULL

File: walkthroughSteps/code-along-mentor-notes.md [hunk 3]
```diff
@@ -3,56 +3,225 @@
...
+      database: params.pathname.split('/')[1],
+      max: process.env.DB_MAX_CONNECTIONS || 2,
+      user: username,
+      password,
+      ssl: params.hostname !== 'localhost',
+    }
+    ```
+    
+    - Use an appropriate number for `max`. More connections mean more memory is used, and too many can crash the database server. Always return connections to the pool (don't block/freeze query callbacks), or the pool will deplete. More connections mean more queries can be run at once and more redundancy incase connections are blocked/frozen.
+    - `ssl` will enable SSL (set to true) if you're not testing on a local machine.
```
Comment (shiryz) [inline walkthroughSteps/code-along-mentor-notes.md:123]: Add that SSL (or secure sockets layer) when set to true ensures you are connecting to the database from a secure server (https)

**Review by shiryz — CHANGES_REQUESTED** [review]
Nothing major it's pretty great :sparkles:

**Comment by macintoshhelper** [comment]
@shiryz I've made the requested changes 👍

**Review by shiryz — APPROVED** [review]
Wicked!

**Review by eliasmalik — APPROVED** [review]

**Pull request merged**